packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -511,23 +502,70 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
     }
   }
 
-  Widget _buildOptionsView(BuildContext context) {
-    return ValueListenableBuilder<BoxConstraints?>(
-      valueListenable: _fieldBoxConstraints,
-      builder: (BuildContext context, BoxConstraints? constraints, Widget? child) {
-        return _RawAutocompleteOptions(
-          fieldKey: _fieldKey,
-          optionsLayerLink: _optionsLayerLink,
-          optionsViewOpenDirection: widget.optionsViewOpenDirection,
-          overlayContext: context,
-          textDirection: Directionality.maybeOf(context),
-          highlightIndexNotifier: _highlightedOptionIndex,
-          fieldConstraints: _fieldBoxConstraints.value!,
-          builder: (BuildContext context) {
-            return widget.optionsViewBuilder(context, _select, _options);
-          },
-        );
-      },
+  // A big enough height for about one item in the default
+  // Autocomplete.optionsViewBuilder. The assumption is that the user likely
+  // wants the list of options to move to stay on the screen rather than get any
+  // smaller than this. Allows Autocomplete to work when it has very little
+  // screen height available (as in b/317115348) by positioning itself on top of
+  // the field, while in other cases to size itself based on the height under
+  // the field.
+  static const double _kMinUsableHeight = kMinInteractiveDimension;
+
+  Widget _buildOptionsView(BuildContext context, OverlayChildLayoutInfo layoutInfo) {
+    if (layoutInfo.childPaintTransform.determinant() == 0.0) {
+      // The child is not visible.
+      return const SizedBox.shrink();
+    }
+    final Size fieldSize = layoutInfo.childSize;
+    final Matrix4 invertTransform = layoutInfo.childPaintTransform.clone()..invert();
+
+    // This may not work well if the paint transform has rotation in it.
+    // MatrixUtils.transformRect returns the bounding rect of the rotated overlay
+    // rect.
+    final Rect overlayRectInField = MatrixUtils.transformRect(
+      invertTransform,
+      Offset.zero & layoutInfo.overlaySize,
```

Comment:
> > Do you know how this approach will be affected by soft keyboard opening and closing
> 
> Correct me if I'm wrong @LongCatIsLooong but wouldn't that just involve incorporating the keyboard insets in `optionsViewMaxHeight`?

@victorsanni I think this is the rect you want to incorporate the keyboard insets to.